code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -29,6 +29,19 @@
 	update_action_button_icons()
 	update_icons()
 
+/mob/living/carbon/Xenomorph/Ravager/handle_status_effects()
+	if(rage) //Rage increases speed, attack speed, armor and fire resistance, and stun/knockdown recovery; speed is handled under movement_delay() in XenoProcs.dm
+		if(world.time > last_rage + 30) //Decrement Rage if it's been more than 3 seconds since we last raged.
+			rage = CLAMP(rage - 5,0,50) //Rage declines over time.
+		AdjustStunned( round(-rage * 0.1,0.01) ) //Recover 0.1 more stun stacks per unit of rage; min 0.1, max 5
+		AdjustKnockeddown( round(-rage * 0.1, 0.01 ) ) //Recover 0.1 more knockdown stacks per unit of rage; min 0.1, max 5
+		adjust_slowdown( round(-rage * 0.1,0.01) ) //Recover 0.1 more stagger stacks per unit of rage; min 0.1, max 5
+		adjust_stagger( round(-rage * 0.1,0.01) ) //Recover 0.1 more stagger stacks per unit of rage; min 0.1, max 5
+		rage_resist = CLAMP(1-round(rage * 0.012,0.01),0.4,1) //+1.2% damage resist per point of rage
+		fire_resist = initial(fire_resist) - round(rage * 0.01,0.01) //+1% fire resistance per stack of rage, max +50%; initial resist is 50%
+		attack_delay = initial(attack_delay) - round(rage * 0.05,0.01) //-0.05 attack delay to a maximum reduction of -2.5
+	. = ..()
```

Comment:
```suggestion
	return ..()
```
If it's at the end, return instead.
This all stacks with other reductions, as you probably know.